crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -151,6 +151,13 @@ impl BaseClient {
         let account_data = &extensions.account_data;
         if !account_data.is_empty() {
             self.handle_account_data(&account_data.global, &mut changes).await;
+            for room_id in account_data.rooms.keys() {
+                if !rooms.contains_key(room_id) {
+                    if let Some(events) = account_data.rooms.get(room_id) {
+                        self.handle_room_account_data(room_id, events, &mut changes).await;
```

Comment:
It would be great if we could get our hands on the RoomInfo here so we can update it based on room account data changes.

Reply:
Perhaps something along these lines instead?

<img width="737" alt="Screenshot 2024-01-19 at 10 08 58" src="https://github.com/matrix-org/matrix-rust-sdk/assets/637564/af5eb9fe-9d47-42bf-a9ba-b53ab82deb88">